apps/ima/behaviours/family-members-locals.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = superclass => class extends superclass {
+  locals(req, res, next) {
+    const superLocals = super.locals(req, res, next);
+    // hide and show fields as necessary on family summary page
+    _.forEach(superLocals.items, i => {
+      _.forEach(i.fields, field => {
+        if (field.field === 'reference-number-option') {
+          if(field.value.includes('do-not-know')) {
+            field.parsed = 'I do not know';
+          } else {
+            field.showInSummary = false;
+          }
+        }
+        if (field.field === 'human-rights-claim-details') {
+          field.showInSummary = false;
+        }
+        if (field.field === 'uan-detail') {
+          if(!field.value) {
+            field.showInSummary = false;
+          } else {
+            field.parsed = field.value;
+          }
+        }
+        if (field.field === 'ho-number-detail') {
+          if(!field.value) {
+            field.showInSummary = false;
+          } else {
+            field.parsed = field.value;
+          }
+        }
+        if (field.field === 'immigration-status-other') {
+          field.showInSummary = false;
+        }
+      });
+    });
+
+    superLocals.items = superLocals.items.map(item => {
+      const immigrationStatus = _.find(item.fields, obj => obj.field === 'ukFamilyMemberImmigrationStatus');
+
+      item.fields = item.fields.map(prop => {
+        const mentalHealthField = prop.field === 'ukFamilyMemberHealthCondition';
+
+        if (_.get(immigrationStatus, 'value') !== 'Dependant on your asylum application'
+          && mentalHealthField) {
+          return null;
+        }
+        if (mentalHealthField) {
+          prop.value = prop.value || 'None';
+        }
+        prop.field += '.summary-heading';
+        return prop;
+      });
+      return item;
+    });
+
+    return superLocals;
+  }
```

Comment:
The function was copied from acq. Would it be better to remove the function when everything has been merged, since the behaviour is used in other stories?

Reply:
Which other story is using this behaviour? I think you can remove this bit from the behaviour now since it is not going to be used by any other field either way ie. there is no mental health question that needs to be displayed on a loop summary page.

Reply:
I've removed this function and re-tested.